sklearn/metrics/_parallel_reductions.pyx

```diff
@@ -721,3 +748,154 @@ cdef class FastSquaredEuclideanArgKmin(ArgKmin):
                       dist_middle_terms[i * Y_c.shape[0] + j] + self.Y_sq_norms[j + Y_start],
                       j + Y_start)
         return 0
+
+
+cdef class RadiusNeighborhood(PairwiseDistancesReduction):
+
+    cdef:
+        DTYPE_t radius
+
+        # Distances metrics compute approximated distance
+        # ("reduced distance" in the original wording),
+        # which are proxies necessitating less computations.
+        # We get the proxy for the radius to be able to compare
+
+        # NOTE: not used for now.
+        DTYPE_t radius_proxy
+
+        # We want resizable buffers which we will
+        # to wrapped within numpy arrays at the end.
+        #
+        # std::vector comes as a handy interface for
+        # efficient resizable buffers.
+        #
+        # Though it is possible to access their buffer
+        # address with std::vector::data, their buffer
+        # can't be stolen: their life-time is tight to
+        # the buffer's.
+        #
+        # To solve this, we allocate dynamically
+        # allocate vectors which won't be
+        # freed, but their buffer eventually will
+        # as the ownership will be transferred to
+        # numpy arrays.
+        #
+        # TODO: Find a proper way to handle this
+        # It's "OK-ish" as numpy arrays are
+        # then responsible for their buffer
+        # lifetime which consist of most of the
+        # vectors actual data (residual metadata
+        # exist, don't account but won't be deleted).
+        # Still, meh.
```

Comment:
Please use your 80 char limit in the comments :)

If you use VS Code you can use https://marketplace.visualstudio.com/items?itemName=konstantin.wrapSelection